src/components/Button.js

```diff
@@ -0,0 +1,27 @@
+const Button = (props) => {
+    const {classes, text, searchCity, getMyGeolocation, toggleModal, clearModalContent} = props;
+
+    const handleClick = (target) => {
+        if (target.classList.contains('button-search')) {
+            searchCity(target);
+        }
+
+        if (target.classList.contains('button-history')) {
+            toggleModal();
+        }
+
+        if (target.classList.contains('button-geolocation')) {
+            getMyGeolocation();
+        }
+
+        if (target.classList.contains('modal__delete')) {
+            clearModalContent();
+        }
+    }
+
+    return(
+        <button className={'button ' + classes} onClick={(e) => handleClick(e.target)}>{text}</button>
+    )
```

Comment:
Компонент стал не реюзабельным, ты должен каждый раз передавать в нужном месте нужный обработчик а не пытаться сделать универсальный
```suggestion
    const {classes, text, onClick} = props;
 
    return(
        <button className={'button ' + classes} onClick={({ target }) => onClick(target)}>{text}</button>
    )
```